From: "Leon M. George" <leon@georgemail.eu>
Date: Wed, 11 Sep 2019 15:22:55 +0200
Subject: [PATCH] hostapd: declare struct wpa_bss early

wps_supplicant.h assumes that 'struct wpa_bss' is forward declared if
CONFIG_WPS is not defined.  With the later inclusion of
600-ubus_support, the issue manifests in warnings like these:

wps_supplicant.h:113:15: warning: 'struct wpa_bss' declared inside parameter list will not be visible outside of this definition or declaration
        struct wpa_bss *bss)
               ^~~~~~~
This patch forward declares 'struct wpa_bss' regardless.

--- a/wpa_supplicant/wps_supplicant.h
+++ b/wpa_supplicant/wps_supplicant.h
@@ -9,6 +9,7 @@
 #ifndef WPS_SUPPLICANT_H
 #define WPS_SUPPLICANT_H
 
+struct wpa_bss;
 struct wpa_scan_results;
 
 #ifdef CONFIG_WPS
@@ -16,8 +17,6 @@ struct wpa_scan_results;
 #include "wps/wps.h"
 #include "wps/wps_defs.h"
 
-struct wpa_bss;
-
 struct wps_new_ap_settings {
 	const char *ssid_hex;
 	const char *auth;
